src/types/store.js

```diff
@@ -19,7 +12,9 @@ import type { State as StateRef } from './reducers';
 export type Action = ActionsRef;
 export type State = StateRef;
 
-export type GetState = ReduxGetState<State>;
-export type ThunkAction<Result> = ReduxThunkAction<State, Action, Result>;
-export type Store = ReduxStore<State, Action>;
-export type Dispatch = ReduxDispatch<State, Action>;
+type ThunkDispatch = <R>(action: ThunkAction<R>) => R;
+type PlainDispatch = (action: Action) => Action;
+export type GetState = () => State;
+export type ThunkAction<R> = (dispatch: Dispatch, GetState) => R;
+export type Dispatch = PlainDispatch & ThunkDispatch;
+export type Store = ReduxStore<State, Action, Dispatch>;
```

Comment:
This is slightly different than what's in https://flow.org/en/docs/react/redux/#toc-typing-redux-thunk-actions but I thought it's nice to type ThunkDispatch and ThunkAction like this so that we properly infer the return result.

Reply:
I'm not sure we need to export all of them; we only need to export ThunkAction and Store, right ?